.github/workflows/coverage_runner.yml

```diff
@@ -0,0 +1,109 @@
+    if: github.event_name == 'push' || needs.check_for_membership.outputs.check-result == 'true' || github.event_name == 'workflow_dispatch'
+    name: Run tests with Python ${{ matrix.python-version }} on ${{ matrix.os }}
+    strategy:
+      matrix:
+        python-version: [ '3.6', '3.10' ]
+        os: [ ubuntu-latest, windows-latest ]
+      fail-fast: false
+    steps:
+      - name: Setup Python
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python-version }}
+      - name: Install JDK
+        uses: actions/setup-java@v2
+        with:
+          distribution: 'adopt'
+          java-version: '8'
+          
+      - name: Checkout code for push event
+        if: github.event_name == 'push'
+        uses: actions/checkout@v2
+
+      - name: Checkout code for PR
+        if: github.event_name == 'pull_request_target'
+        uses: actions/checkout@v2
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
+          
+      - name: Checkout repository to fetch PR coming from community
+        if: github.event_name == 'workflow_dispatch'
+        uses: actions/checkout@v2
+        
+      - name: Fetch PR coming from community
+        if: github.event_name == 'workflow_dispatch'
+        run: |
+          git fetch origin pull/${{github.event.inputs.pr_number }}/head:temp_branch
+          git checkout temp_branch
+          
+      - name: Install dependencies
+        run: |
+          pip install -r requirements-test.txt
+      - name: Run tests
+        env:
+          HAZELCAST_ENTERPRISE_KEY: ${{ secrets.HAZELCAST_ENTERPRISE_KEY }}
+        run: python run_tests.py
+            
+      - name: Publish results to Codecov for PR coming from hazelcast organization
+        if: ${{ matrix.python-version == '3.10' && matrix.os == 'ubuntu-latest' &&  github.event_name == 'pull_request_target' }}
+        uses: codecov/codecov-action@v1
+        with:
+          files: ./coverage.xml
+          override_pr: ${{ github.event.pull_request.number }}
+          
+      - name: Publish results to Codecov for Push
+        if: ${{ matrix.python-version == '3.10' && matrix.os == 'ubuntu-latest' &&  github.event_name == 'push' }}
+        uses: codecov/codecov-action@v1
+        with:
+          files: ./coverage.xml
+          
+      - name: Publish result to Codecov for PR coming from community
+        if: ${{ matrix.python-version == '3.10' && matrix.os == 'ubuntu-latest' && github.event_name == 'workflow_dispatch' }}
+        uses: codecov/codecov-action@v1
+        with:
+          files: ./coverage.xml
+          override_pr: ${{ github.event.inputs.pr_number }}
+          
+      - name: Upload remote controller logs if test run fails
+        uses: actions/upload-artifact@v2
+        if: failure()
+        with:
+          name: rc-logs-${{ matrix.python-version }}-${{ matrix.os }}
+          path: |
+            rc_stderr.log
+            rc_stdout.log
```

Comment:
Not a big issue we can make the newlines between steps consistent. Right now some of them has a newline some don't and I don't see a logical grouping either.

Reply:
Fixed at [https://github.com/hazelcast/hazelcast-python-client/pull/539/commits/7cb06a7488d96c1f42d758b6ac12900be23a3051](url)